src/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs

```diff
@@ -143,11 +143,23 @@ public void TimesIncludeMillisecondPart_Windows()
         public void TimesIncludeMillisecondPart_OSX()
         {
             T item = GetExistingItem();
-            Assert.All(TimeFunctions(), (function) =>
+            string driveFormat = new DriveInfo(Path.GetTempPath()).DriveFormat;
+            if (driveFormat.Equals("hfs", StringComparison.InvariantCultureIgnoreCase))
             {
-                DateTime time = function.Getter(item);
-                Assert.Equal(0, time.Millisecond);
-            });
+                Assert.All(TimeFunctions(), (function) =>
+                {
+                    DateTime time = function.Getter(item);
+                    Assert.Equal(0, time.Millisecond);
+                });
+            }
+            else
```

Comment:
I actually think it's not worth bothering doing anything on hfs. Just make the test skip if driveFormat != apfs.